srml/executive/src/lib.rs

```diff
@@ -14,7 +14,55 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
-//! Executive: Handles all of the top-level stuff; essentially just executing blocks/extrinsics.
+//! # Executive Module
+//!
+//! The executive module acts as the orchestration layer for the runtime. It dispatches incoming
+//! extrinsic calls to the respective modules in the runtime.
+//!
+//! ## Overview
+//!
+//! The executive module is coupled with the [system module](../srml_system/index.html) and lets the runtime
+//! communicate with the SRML. It is unique from other modules in that it does not incorporate the `decl_event!`,
+//! `decl_storage!`, or `decl_module!` macros.
+//!
+//! The executive module provides functions to:
+//!
+//! - Check transaction validity.
+//! - Initialize a block.
+//! - Apply extrinsics.
+//! - Execute a block.
+//! - Finalize a block.
+//! - Start an off-chain worker.
+//!
+//! ### Implementations
+//!
+//! The executive module provides the following implementations:
+//!
+//! - `ExecuteBlock`: Trait that can be used to execute a block.
+//! - `Executive`: Type that can be used to make the SRML available from the runtime.
+//!
+//! ## Usage
+//!
+//! The default Substrate node template declares the [`Executive`](./struct.Executive.html) type in its library.
+//!
+//! ### Example
+//!
+//! `Executive` type declaration from the node template.
+//!
+//! ```ignore
+//! /// Executive: handles dispatch to the various modules.
+//! pub type Executive = executive::Executive<Runtime, Block, Context, Balances, AllModules>;
+//! ```
+//!
+//! ## Genesis Config
+//!
+//! The executive module is not dependent on the genesis config.
+//!
+//! ## Related Modules
+//!
+//! The executive module depends on the [`system`](../srml_system/index.html)  and
+//! [`srml_support`](../srml_support/index.html) modules as well as Substrate Core
+//! libraries and the Rust standard library.
```

Comment:
I guess this section is a bit redundant in this case. We have mentioned the system module before and support macros are not being mentioned as related modules in other docs. So, can we get rid of it?